src/format/parse.rs

```diff
@@ -224,6 +225,263 @@ fn parse_rfc3339<'a>(parsed: &mut Parsed, mut s: &'a str) -> ParseResult<(&'a st
+            if extended_format {
+                s = scan::char(s, b'-')?;
+            }
+            parsed.set_day(try_consume!(scan::number(s, 2, 2)))?;
+        }
+    }
+    let format = if extended_format { Iso8601Format::Extended } else { Iso8601Format::Basic };
+    Ok((s, format))
+}
+
+fn parse_iso8601_year(mut s: &str) -> ParseResult<(&str, i64)> {
+    match s.as_bytes().first() {
+        Some(sign) if sign == &b'-' || sign == &b'+' => {
+            // expanded representation
+            let negative = sign == &b'-';
+            s = &s[1..];
+            let mut digits = s.as_bytes().iter().take_while(|c| c.is_ascii_digit()).count();
+            if let Some(&b'-' | &b'W') = s.as_bytes().get(digits) {
+                // The date format is either an extended format with `-` as seperator between date
+                // fields, or it is a week date in basic format. In both cases all counted digits
+                // belong to the year.
+                if digits < 4 {
+                    return Err(TOO_SHORT);
+                }
+            } else if digits == 7 {
+                digits -= 3; // must be the format ±YYYYDDD
+            } else if digits > 7 {
+                // The basic format with expanded representation of a calendar date (±Y̲YYYYMMDD)
+                // and ordinal date (±Y̲YYYYDDD) are ambiguous. In this case we assume a calendar
+                // date, where the last 4 digits are for the month and day.
+                digits -= 4;
+            } else {
+                return Err(TOO_SHORT);
+            }
+            let (s, year) = scan::number(s, 4, digits)?;
+            Ok((s, if negative { -year } else { year }))
+        }
+        Some(_) => scan::number(s, 4, 4),
+        None => Err(TOO_SHORT),
+    }
+}
+
+/// The ISO 8601 time format has a basic and an extended format, representations with reduced
+/// accuracy, and representations with a decimal fraction:
+///
+/// |                                   | basic format | extended format |
+/// |-----------------------------------|--------------|-----------------|
+/// | complete representation           | hhmmss       | hh:mm:ss        |
+/// | reduced accuracy: hour and minute | hhmm         | hh:mm           |
+/// | reduced accuracy: hour            | hh           |                 |
+/// | decimal fraction of the second    | hhmmss,ss̲    | hh:mm:ss,ss̲     |
+/// | decimal fraction of the minute    | hhmm,mm̲      | hh:mm,mm̲        |
+/// | decimal fraction of the hour      | hh,hh̲        | hh,hh̲           |
+///
+/// As decimal sign is both `,` and `.` are allowed, with `,` being preferred.  A decimal fraction
+/// must have at least one digit. The standard puts no limit on the number of digits.
+///
+/// Midnight can be represented with both `00:00` (at the start of the day) and `24:00` (at the end
+/// of the calendar day).
+pub(crate) fn parse_iso8601_time<'a>(
```

Comment:
Please describe the meaning of the returned tuple values within the `ParseResult`.  It's not clear to me what these represent (except for the `Iso8601Format`).